### PR TITLE
[4.0] Breadcrumb menu item context

### PR DESCRIPTION
**Pull request by Tchangue**
Pull Request for Issue [here](https://github.com/joomla/joomla-cms/pull/21361)

### Summary of Changes
For each  menu item we can now see the full path beninning from its root parent.

### Testing Instructions
You can create two menu items and one of them must be the kind item of the other.

### Expected result
The path of items must be displayed. For root parent item only the item is showed.

### Actual result
The path of items are displayed. For root parent item only the item is showed.

### CODE REVIEW

**Comment by brianteeman** [comment]
PR was submitted against the wrong branch (staging) I have changed it to 4.0-dev

File: administrator/components/com_menus/Model/ItemsModel.php [hunk 1]
```diff
@@ -28,8 +29,8 @@ class ItemsModel extends ListModel
 	/**
 	 * Constructor.
 	 *
-	 * @param   array                $config   An optional associative array of configuration settings.
-	 * @param   MVCFactoryInterface  $factory  The factory.
+	 * @param   array               $config  An optional associative array of configuration settings.
```
Comment (Quy) [inline administrator/components/com_menus/Model/ItemsModel.php]: Please revert. Per the [Joomla Coding Standard](https://developer.joomla.org/coding-standards/docblocks.html):

```
Note: Commonly a line after the tag @param consists of the following three parts in order of appearance:

    variable type (There must be 3 spaces before variable type.)
    variable name (There must be 2 spaces after the longest type.)
    variable description (There must be 2 spaces after the longest variable name.)

If there are more than one @param the type, names and description have to be aligned.
```

File: administrator/components/com_menus/Model/ItemsModel.php [hunk 2]
```diff
@@ -586,4 +587,26 @@ public function getItems()
...
+     * Search the parent item id to select the parent item name
+     * Breadcrumb
+     * */
+	public function searchParentItem($value)
+	{
+		$db    = Factory::getDbo();
+		$query = $db->getQuery(true);
+		$query->select("*")->from("#__menu")->where("id = " . $value);
+		$result = $db->setQuery($query)->loadObject();
+		if (empty($result))
```
Comment (Quy) [inline administrator/components/com_menus/Model/ItemsModel.php]: Add blank line above.

File: administrator/components/com_menus/tmpl/items/default.php [hunk 3]
```diff
@@ -38,68 +38,68 @@
...
-$assoc   = Associations::isEnabled() && $this->state->get('filter.client_id') == 0;
+$assoc = Associations::isEnabled() && $this->state->get('filter.client_id') == 0;
 
 ?>
 <?php // Set up the filter bar. ?>
 <form action="<?php echo Route::_('index.php?option=com_menus&view=items'); ?>" method="post" name="adminForm"
       id="adminForm">
-	<div class="row">
-		<div id="j-sidebar-container" class="col-md-2">
+    <div class="row">
```
Comment (Quy) [inline administrator/components/com_menus/tmpl/items/default.php:47]: Use tabs and not spaces to indent.

**Comment by rdeutz** [comment]
they are code style issues you have to fix, see http://ci.joomla.org/joomla/joomla-cms/8441/8

**Comment by ghost** [comment]
@Tchangue thanks for your work. Closed in favor of #26187

**Pull request closed without merging**